### ISSUE TITLE
rate_limited status and nextAvailableAt option set before the quota is reached

### ISSUE THREAD

**Issue by samhibberd** [matomo-org/plugin-GoogleAnalyticsImporter#430]
Running into some strange behavior with a historic import of around 5 years of GA (UA) data into a self hosted mamoto instance, the issue was inconsistent initially, getting generic google errors (as described in some other issues) but it appears (after running through and setting things up again) to be more consistently a `rate_limited` error. But it looks to come in before the quota is met, and then it looks to set the nextAvailableAt database option too far in the future.

We have setup the import as per all the docs and we have the archiving cron setup. 

As I understand it the daily quota for the Analytics Reporting API is 50k, but our daily quota usage is never reported above 20% 10k requests. Has anything changed? Is the reporting accurate?

I also have Quota alerts setup in google (triggered at 90%) and they have not been triggered once 🤷 

_NB. Running the import-reports console command in debug mode (-vvv) does occasionally report a quota issue, but not always._

Any thoughts?
Happy to provide more detail on our setup if required.

**Comment by AltamashShaikh** [comment]
@samhibberd As per this [doc](https://developers.google.com/analytics/devguides/reporting/core/v4/limits-quotas)
“10,000 requests per view (profile) per day (cannot be increased)”

This explains why its 10k

**Comment by samhibberd** [comment]
Ahhh, missed that, quite misleading in the console and makes their alerts pretty pointless:

<img width="1019" alt="Screenshot 2023-07-20 at 17 08 25" src="https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/assets/1846063/748515b3-42e6-485e-8af6-ef9ebccdf69c">

And the `nextAvailableAt` option, currently being set for tomorrow at 1pm (UTC+1) but would have expect it to be set at midnight tonight, when the quota resets?

**Comment by AltamashShaikh** [comment]
@samhibberd Are you viewing this value in DB or interface ?
In Database the timestamp is store relative to timezone - https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/blob/4.x-dev/Google/GoogleAnalyticsQueryService.php#L322

**Comment by samhibberd** [comment]
The database. Feels like some process is getting confused, just checking now and the timestamp is 1689984000 (which is tomorrow), our server is UTC, we are at UTC+1 and the last data was imported yesterday, so thats an entire day skipped, where there is quota available. I can kick start it by removing that option from the database manually and triggering a restart in the mamoto dashboard.

When I thought the quota was 50k, not 10k, I would try to restart manually in the mamoto dashboard (or via the cli) would this process kick the nextAvailableAt timestamp back another day?

**Comment by AltamashShaikh** [comment]
@samhibberd You mean the `nextAvailableAt` is updated without running any imports for today ?
You can delete the option and it should run automatically next hour, if you have `cron:archive` setup

**Comment by samhibberd** [comment]
Yes exactly, the date was updated such that it missed today.

I have deleted and it is now running, I will wait and check what the `nextAvailableAt` option is set to when todays quota is used up.

**Comment by AltamashShaikh** [comment]
@samhibberd please update us may be some issue with the timezone check

**Comment by samhibberd** [comment]
I've just checked, quota has been used up and the `nextAvailableAt` date is correctly set for first thing tomorrow in UTC `Saturday, July 22, 2023 12:00:00 AM`. 

I have tried a few things to recreate the issue (none restart the task due to the quota being exceeded which is expected):

1) Restarting manually via plugin settings - `nextAvailableAt` correctly unaffected.
2) Restarting manually via cli - `nextAvailableAt` correctly unaffected (although the warning is output twice ~~and the date is incorrect~~):
<img width="1043" alt="Screenshot 2023-07-21 at 15 58 11" src="https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/assets/1846063/743aa94c-c52e-41f5-8b60-edddc3dc6475">

3) Deleting the option and then restarting via plugin settings or cli - `nextAvailableAt` correctly unaffected.

So rather confused how this could have happened 🤷
